compliance_checker/tests/test_cf.py

```diff
@@ -1010,3 +1014,197 @@ def test_units_temporal(self):
         self.assertTrue(units_temporal('hours since 2000-01-01'))
         self.assertFalse(units_temporal('hours'))
         self.assertFalse(units_temporal('days since the big bang'))
+
+
+class TestCF17(TestCF16):
+    """Extends the CF 1.6 tests. Most of the tests remain the same."""
+
+    def setUp(self):
+        '''Initialize a CF17Check object.'''
+
+        self.cf = CF17Check()
+
+    def test_check_actual_range(self):
+        """Test the check_actual_range method works as expected"""
+
+        # using a with block closes the ds; for checks operating on the data, we need
+        # to intialize and then manually close
+
+        dataset = MockTimeSeries()
+        dataset.createVariable('a', 'd', ('time',)) # dtype=double, dims=time
+        # test that if the variable doesn't have an actual_range attr, no score 
+        result = self.cf.check_actual_range(dataset)
+        assert result == []
+        dataset.close()
+
+        # NOTE this is a data check
+        # if variable values are equal, actual_range should not exist
+        dataset = MockTimeSeries()
+        dataset.createVariable('a', 'd', ('time',)) # dtype=double, dims=time
+        dataset.variables["a"][0:500] = 0 # set all 500 vals to 0
+        dataset.variables["a"].setncattr("actual_range", [1, 1]) # shouldn't exist
+        result = self.cf.check_actual_range(dataset)
+        score, out_of, messages = self.get_results(result)
+        assert score < out_of
+        assert len(messages) == 1
+        assert messages[0] == u"\"a\"'s values are all equal; actual_range shouldn't exist"
```

Comment:
Python 3 uses Unicode strings by default, which are in UTF-8.  UTF-8 as an encoding has the property that the ASCII character set is defined in one byte, and is so is equivalent to the first 128 characters in ASCII.  Thus I don't think the `u` isn't strictly necessary as the bytes str (Python 2) and unicode str (Python 3) should have the same representation in terms of the byte content.